lib/bandit.ex

```diff
@@ -321,7 +321,9 @@ defmodule Bandit do
     handler_options = %{
       plug: plug,
       handler_module: Bandit.InitialHandler,
-      opts: %{http_1: http_1_options, http_2: http_2_options, websocket: websocket_options}
+      opts: %{http_1: http_1_options, http_2: http_2_options, websocket: websocket_options},
+      http_1_enabled: Keyword.get(http_1_options, :enabled, true),
+      http_2_enabled: Keyword.get(http_2_options, :enabled, true)
     }
```

Comment:
Do they need to stay in the original set of options then?

```suggestion
{http_1_enabled, http_1_options} = Keyword.pop(http_1_options, :enabled, true)
{http_2_enabled, http_2_options} = Keyword.pop(http_2_options, :enabled, true)

handler_options = %{
  plug: plug,
  handler_module: Bandit.InitialHandler,
  opts: %{http_1: http_1_options, http_2: http_2_options, websocket: websocket_options},
  http_1_enabled: http_1_enabled,
  http_2_enabled: http_2_enabled
}
```

Reply:
I am not quite sure. The `opts` get passed down to the `Adapter` but it seem the `enabled` field is only used for websockets during upgrade attempts and not for http1 or http2, so we could safely remove them from the `opts`.